### PR TITLE
Remove CreateStaticHandlers from VM interface

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

Removes static apis. These are currently exposed to let users interact with a VM without it being initialized, but we can avoid this complexity by replacing a static api with some static function that isn't bound to the VM. 

## How this works

Removes the static handlers from the VM interface.

The static api code still exists to create genesis files/etc. Follow-up [issue](https://github.com/ava-labs/avalanchego/issues/2613) will be done to refactor this into static functions instead of having the useless static api struct.

## How this was tested

CI

### CODE REVIEW

File: proto/vm/vm.proto [hunk 1]
```diff
@@ -21,13 +21,6 @@ service VM {
   rpc Shutdown(google.protobuf.Empty) returns (google.protobuf.Empty);
   // Creates the HTTP handlers for custom chain network calls.
   rpc CreateHandlers(google.protobuf.Empty) returns (CreateHandlersResponse);
-  // Creates the HTTP handlers for custom VM network calls.
-  //
-  // Note: RPC Chain VM Factory will start a new instance of the VM in a
-  // separate process which will populate the static handlers. After this
-  // process is created other processes will be created to populate blockchains,
-  // but they will not have the static handlers be called again.
-  rpc CreateStaticHandlers(google.protobuf.Empty) returns (CreateStaticHandlersResponse);
```
Comment (StephenButtolph) [inline proto/vm/vm.proto:30]: I think we should be removing the `CreateStaticHandlersResponse` type as well

File: vms/registry/vm_registerer.go [hunk 2]
```diff
@@ -70,75 +60,16 @@ func (r *vmRegisterer) register(ctx context.Context, pathAdder server.PathAdder,
...
 	defaultEndpoint := path.Join(constants.VMAliasPrefix, vmID.String())
-
-	if err := r.createStaticEndpoints(pathAdder, handlers, defaultEndpoint); err != nil {
-		return err
-	}
 	urlAliases, err := r.getURLAliases(vmID, defaultEndpoint)
 	if err != nil {
 		return err
 	}
 	return pathAdder.AddAliases(defaultEndpoint, urlAliases...)
```
Comment (StephenButtolph) [inline vms/registry/vm_registerer.go:70]: This can all be removed as well right?

File: vms/registry/vm_registerer.go [hunk 3]
```diff
@@ -70,75 +60,16 @@ func (r *vmRegisterer) register(ctx context.Context, pathAdder server.PathAdder,
...
-				defaultEndpoint,
-				extension,
-				err,
-			)
-		}
-	}
-	return nil
-}
-
 func (r vmRegisterer) getURLAliases(vmID ids.ID, defaultEndpoint string) ([]string, error) {
```
Comment (StephenButtolph) [inline vms/registry/vm_registerer.go:73]: I don't think this function should exist anymore.

File: vms/registry/vm_registerer.go [hunk 4]
```diff
@@ -70,75 +60,16 @@ func (r *vmRegisterer) register(ctx context.Context, pathAdder server.PathAdder,
 	if err := r.config.VMManager.RegisterFactory(ctx, vmID, factory); err != nil {
 		return err
 	}
-	handlers, err := r.createStaticHandlers(ctx, vmID, factory)
-	if err != nil {
-		return err
-	}
 
 	// all static endpoints go to the vm endpoint, defaulting to the vm id
 	defaultEndpoint := path.Join(constants.VMAliasPrefix, vmID.String())
```
Comment (StephenButtolph) [inline vms/registry/vm_registerer.go:65]: I think `constants.VMAliasPrefix` should be unused after this PR

**Review by dhrubabasu — APPROVED** [review]

File: scripts/mocks.mockgen.txt [hunk 5]
```diff
@@ -38,7 +38,6 @@ github.com/ava-labs/avalanchego/vms/proposervm/scheduler=Scheduler=vms/proposerv
 github.com/ava-labs/avalanchego/vms/proposervm/state=State=vms/proposervm/state/mock_state.go
 github.com/ava-labs/avalanchego/vms/proposervm=PostForkBlock=vms/proposervm/mock_post_fork_block.go
 github.com/ava-labs/avalanchego/vms/registry=VMGetter=vms/registry/mock_vm_getter.go
-github.com/ava-labs/avalanchego/vms/registry=VMRegisterer=vms/registry/mock_vm_registerer.go
```
Comment (StephenButtolph) [inline scripts/mocks.mockgen.txt:41]: I think we still need to delete the generated file from this.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**